app/models/subscription.rb

```diff
@@ -51,11 +53,30 @@ def new_comment_by_someone_else?
   end
 
   def last_author
-    (discussion.comments.last || discussion).author
+    (last_comment || discussion).author
   end
 
-  def notify
-    NotificationsMailer.new_comment(id).deliver_later
+  def last_comment
+    discussion.comments.last
+  end
+
+  def notify_new_discussion
+    return unless new?
+    user.notifications.new_discussion(discussion)
```

Comment:
that was the idea, and I believe the first version was similar to that, but it would require me to pass the user_id as a parameter as well. This way that's implicit